### PR TITLE
Exposing the simulation results through public functions in sim-lib

### PR DESCRIPTION
**Pull request by bjohnson5**
Closes Issue #196. Adds public functions to sim-lib that let a library user get the current simulation statistics.

### CODE REVIEW

File: sim-lib/src/lib.rs [hunk 1]
```diff
@@ -770,7 +781,7 @@ impl Simulation {
             }
         });
 
-        let result_logger = Arc::new(Mutex::new(PaymentResultLogger::new()));
+        let result_logger = self.results.clone();
```
Comment (carlaKC) [inline sim-lib/src/lib.rs:784]: Do we need both `let result_logger = self.results.clone();` and `let result_logger_clone = result_logger.clone();` below? Seems like we could reduce to a single clone?

**Review by carlaKC — APPROVED** [review]
lgtm, one nit

File: sim-lib/src/lib.rs (hunk 1 above)
Comment (bjohnson5) [reply sim-lib/src/lib.rs:784]: Hm. I think you need a results logger in the `run_results_logger` spawned task and the `consume_simulation_results` spawned task as well as the main simulation thread. So the Arc should have a reference count of 3? Which means you need to clone it twice. Am I understanding that correctly? Very possible I am missing something.

Comment (carlaKC) [reply sim-lib/src/lib.rs:784]: Nope you're totally right - missed `consume_simulation_results` below!

**Pull request merged**